callstats-java-sdk/src/main/java/io/callstats/sdk/CallStats.java

```diff
@@ -148,43 +156,59 @@ public void sendCallStatsBridgeStatusUpdate(BridgeStatusInfo bridgeStatusInfo) {
 					logger.debug("Response "+response.toString()+":"+responseStatus);
-					BridgeStatusUpdateResponse eventResponseMessage;
-					try {
-						String responseString = EntityUtils.toString(response.getEntity());
-						eventResponseMessage  = gson.fromJson(responseString,BridgeStatusUpdateResponse.class);	
-					} catch (ParseException e) {						
-						logger.error("ParseException "+e.getMessage(),e);
-						throw new RuntimeException(e);
-					} catch (IOException e) {
-						logger.error("IO Execption "+e.getMessage(),e);
-						throw new RuntimeException(e);					
-					} catch (JsonSyntaxException e) {
-						logger.error("Json Syntax Exception "+e.getMessage(),e);
-						e.printStackTrace();
-						throw new RuntimeException(e);
-					}
-					if(responseStatus == 200) {
+					if(responseStatus == CallStatsResponseStatus.RESPONSE_STATUS_SUCCESS) {
+						BridgeStatusUpdateResponse eventResponseMessage;
+						try {
+							String responseString = EntityUtils.toString(response.getEntity());
+							eventResponseMessage  = gson.fromJson(responseString,BridgeStatusUpdateResponse.class);	
+						} catch (ParseException e) {						
+							logger.error("ParseException "+e.getMessage(),e);
+							throw new RuntimeException(e);
+						} catch (IOException e) {
+							logger.error("IO Execption "+e.getMessage(),e);
+							throw new RuntimeException(e);					
+						} catch (JsonSyntaxException e) {
+							logger.error("Json Syntax Exception "+e.getMessage(),e);
+							e.printStackTrace();
+							throw new RuntimeException(e);
+						}
 						logger.debug("Response status is "+eventResponseMessage.getStatus()+":"+eventResponseMessage.getReason());
 						if (eventResponseMessage.getStatus().equals("Error") && eventResponseMessage.getReason().contains(CallStatsErrors.INVALID_TOKEN_ERROR.getReason())) {
 							//logger.debug("Response status is "+eventResponseMessage.getStatus()+":"+eventResponseMessage.getReason());
 							bridgeKeepAliveManager.stopKeepAliveSender();
 							authenticator.doAuthentication();
-						}						
+						}
+						httpClient.setDisrupted(false);
+						sendCallStatsBridgeStatusUpdateFromQueue();
+					} else {
+						httpClient.setDisrupted(true);
 					}
 				}
 				
 				public void onFailure(Exception e) {
 					logger.error("Response exception"+e.getMessage(),e);
+					httpClient.setDisrupted(true);
 				}
 				
 			});	
 		} else {
 			// TODO retransmission queue
-			throw new UnsupportedOperationException("queueing not implemented yet");
+			//throw new UnsupportedOperationException("queueing not implemented yet");
+			bridgeStatusInfoQueue.push(bridgeStatusInfo);
+		}
+	}
+	
+	
+	public synchronized void sendCallStatsBridgeStatusUpdateFromQueue() {
+		if (bridgeStatusInfoQueue.getLength() < 1)
+			return;
+		
+		while (bridgeStatusInfoQueue.getLength() > 0) {
+			BridgeStatusInfo bridgeStatusInfo = bridgeStatusInfoQueue.pop();
+			sendCallStatsBridgeStatusUpdate(bridgeStatusInfo);
```

Comment:
Do we need any message pacing here? Can we cause message storm in the backend by sending too many events simultaneously?